latex/programming_interface.tex

```diff
@@ -3770,16 +3779,19 @@ \subsection{Stream class interface}
   \addRow
     {template <typename T> const stream\& operator<<(const stream\& os, const T \&rhs)}
     {
-      Outputs any valid values (see~\ref{table.operands.stream}) as a stream of characters and applies any valid manipulator (see~\ref{table.manipulators.stream}) to the current statements.
+      Outputs any valid values (see~\ref{table.operands.stream}) as a stream of characters and applies any valid manipulator (see~\ref{table.manipulators.stream}) to the current stream.
     }
 \completeTable
 %-------------------------------------------------------------------------------------------
 
 \subsection{Synchronization}
 
-An instance of the SYCL \codeinline{stream} class is required to synchronize with the host, and must output everything that is streamed to it via the \codeinline{operator<<()} operator (that doesn't exceed the \codeinline{maxStatementSize} or \codeinline{bufferSize} limits) within a SYCL kernel function by the time that the event associated with a command group submission enters the completed state. The point at which this synchronization occurs and the method by which this synchronization is performed are implementation defined. For example it is valid for an implementation to use \codeinline{printf()}.
+An instance of the SYCL \codeinline{stream} class is required to synchronize with the host, and must output everything that is streamed to it via the \codeinline{operator<<()} operator (that doesn't exceed the \codeinline{workItemBufferSize} or \codeinline{totalBufferSize} limits) within a SYCL kernel function by the time that the event associated with a command group submission enters the completed state. The point at which this synchronization occurs and the method by which this synchronization is performed are implementation defined. For example it is valid for an implementation to use \codeinline{printf()}.
```

Comment:
When specifying what is required to be synchronised with the host, we should specify that it's everything that is streamed via `operator<<` before performing a flush. Clarifying that if you do not flush, there is no guarantee that anything is output.

Reply:
I added the wording. Note that we've said that there is an implicit flush on kernel completion, defined in the "Implicit flush" section near the end of the stream section.  We believe that the implicit flush on kernel completion is important for usability.

Reply:
I agree this would be semantically preferable, though this requirement could infer the implementation strategy because in order for this to happen the per-work-item buffer would have to outlive the kernel function. Which either means copying back the collective buffer that contains each work item's buffer or inserting an implicit flush at the end of the kernel function.

Reply:
Yes, understood.  Are you suggesting that there should be no implicit flush on kernel completion because it constrains implementations too much, or just noting that there are implementation ramifications?